Keep unnamed imports

GregoryVigoTorres opened the pull request:
Fixes issue #10

GregoryVigoTorres committed:
Fix from import with multiple sub-packages

GregoryVigoTorres committed:
Do not import bar as bar

GregoryVigoTorres committed:
flake8

Removed test

keyan commented:
Hello Gregory, thanks for taking the time to submit this PR. I made a couple comments within the commits. Also, could you please resolve any merge conflicts?

keyan commented on tests/test_import_migration.py:
```diff
@@ -14,7 +14,13 @@ def test_simple_from_import():
     red = RedBaron("from flask.ext.foo import bar")
     output = migrate.fix_tester(red)
-    assert output == "from flask_foo import bar as bar"
+    assert output == "from flask_foo import bar"
```
It may be easier to understand if these test changes (here and L20) go in 99f9ff3.

keyan commented on flask_ext_migrate/__init__.py:
```diff
@@ -33,33 +33,44 @@ def fix_from_imports(red):
+                    tars_str += i.value
+                    if (i.type == 'name_as_name' 
+                        and i.next 
+                        and i.next.type != ('right_parenthesis')):
```
I'd prefer we define a constant at the beginning of the module so it is clear what the string "right_parenthesis" is and where it is defined.

keyan commented on flask_ext_migrate/__init__.py:
```diff
@@ -33,33 +33,44 @@ def fix_from_imports(red):
+                         % (modules_str, tars_str))
+
+        elif len(modules) == 2:
+            # simple from import
```
I'm not sure what this comment means, it can probably be omitted, unless you'd rather make it more clear.

keyan commented on flask_ext_migrate/__init__.py:
```diff
@@ -33,33 +33,44 @@ def fix_from_imports(red):
-                                 % (package, module_string))
+
+        if len(modules) >= 3:
+            tars_str = ''
```
Could we come up with a better variable name, here and for other lines where we are using `tar`?

GregoryVigoTorres committed:
Refactor fix_from_imports

GregoryVigoTorres commented:
Sorry for being disorganized with my pull requests: I don't have much github experience and I've been really busy with other things. I hope this is the right way to do this.

The last commit addresses most of your comments, and I've refactored the code to make it a little cleaner.

`right-parenthesis` on line 66 is the RedBaron node type, like `name-as-name`on the line above.

untitaker commented:
Could you rebase against master, or merge master into your branch? There are some conflicts that need to be resolved.

GregoryVigoTorres committed:
Merge remote-tracking branch 'origin' into keep_unnamed_imports

keyan commented:
Hi @GregoryVigoTorres, given your comment I realize it is possible that it is not clear what Markus and I are asking for.

You'll need to pull the latest `master` and rebase onto it. There will be merge conflicts that you will have to resolve manually:

First, add the main repo as a remote if you haven't already

```
$ git remote add upstream git@github.com:pallets/flask-ext-migrate.git
```

Now get the latest changes from the main repo:

```
$ git checkout master
$ git pull upstream master
```

Now back on your branch, rebase onto the latest master:

```
$ git checkout keep_unnamed_imports
$ git rebase -i upstream/master
```

The rebase will stop midway and require you to fix the conflicts, refer to the [github documentation regarding this](https://help.github.com/articles/resolving-merge-conflicts-after-a-git-rebase/) for the proper procedure.

Once you have completed the rebase, you will need to force push in order to update your remote repo:

```
$ git push -f
```

untitaker commented:
Ah sorry, I didn't properly read the existing conversation.

GregoryVigoTorres commented:
hello Keyan,

Yeah... I didn't really understand what you meant because I don't have 
write access to the pallets/master so I was thinking 'How could I merge 
onto it?'

My master is up to date with upstream/master. I merged my master onto my 
branch, resolving conflicts, and pushed my merge commit.

Is that going to work?

On 08/19/2016 05:10 PM, Keyan Pishdadian wrote:

> Hi @GregoryVigoTorres https://github.com/GregoryVigoTorres, given 
> your comment I realize it is possible that it is not clear what Markus 
> and I are asking for.
> 
> You'll need to pull the latest |master| and rebase onto it. There will 
> be merge conflicts that you will have to resolve manually:
> 
> First, add the main repo as a remote if you haven't already
> 
> |$ git remote add upstream git@github.com:pallets/flask-ext-migrate.git |
> 
> Now get the latest changes from the main repo:
> 
> |$ git checkout master $ git pull upstream master |
> 
> Now back on your branch, rebase onto the latest master:
> 
> |$ git checkout keep_unnamed_imports $ git rebase -i upstream/master |
> 
> The rebase will stop midway and require you to fix the conflicts, 
> refer to the github documentation regarding this 
> https://help.github.com/articles/resolving-merge-conflicts-after-a-git-rebase/ 
> for the proper procedure.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> https://github.com/pallets/flask-ext-migrate/pull/11#issuecomment-241045183, 
> or mute the thread 
> https://github.com/notifications/unsubscribe-auth/AM6RLCgSCb1jcylM4ZjUWOKhNF8afrpEks5qhcdxgaJpZM4JZlAB.

keyan commented:
Gregory, that sounds work too. Try pushing it here and let us see.

Also, if you are on IRC, hop into the #pocoo channel on freenode if you have any more questions for me.

GregoryVigoTorres commented:
OK

It's been pushed.

commit a7c4f4c says Merge remote-tracking branch 'origin'

but origin is the same as upstream

On 08/19/2016 05:33 PM, Keyan Pishdadian wrote:

> Gregory, that sounds work too. Try pushing it here and let us see.
> 
> Also, if you are on IRC, hop into the #pocoo channel on freenode if 
> you have any more questions for me.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> https://github.com/pallets/flask-ext-migrate/pull/11#issuecomment-241051454, 
> or mute the thread 
> https://github.com/notifications/unsubscribe-auth/AM6RLF9qQ07W3LM0mbA3tYfXSu2pHvqPks5qhcy2gaJpZM4JZlAB.

keyan commented:
Hi Gregory, thank you for your patience. The build is still failing but it is not related to your change, it should be resolved by #14.

I made a couple more comments on the changes. Also, let me know if you don't have enough time to get to it, I'm happy to help as well.

keyan commented on flask_ext_migrate/__init__.py:
```diff
@@ -42,33 +42,36 @@ def fix_from_imports(red):
-        if (len(modules) < 2 or
-                modules[0].value != 'flask' or modules[1].value != 'ext'):
-                    continue
+        if modules[0].value != 'flask' and modules[1].value != 'ext':
```
I think this may break simple imports such as `import requests`, could you add a test for that?

edit: Hmm, not really, I glazed over L41. This should be fine. I might add some simple tests for sanity later though. 👍

keyan commented:
I think this is ready to go 👍
